src/edu/washington/escience/myria/operator/agg/LongAggregator.java

```diff
@@ -63,42 +61,6 @@ public void addRow(
   }
 
   @Override
-  public List<Column<?>> emitOutput(final TupleBatch tb) {
-    switch (aggOp) {
-      case COUNT:
-      case SUM:
-      case MAX:
-      case MIN:
-        return ImmutableList.of(tb.getDataColumns().get(stateCols[0]));
-      case AVG:
-        {
-          ReadableColumn sumCol = tb.asColumn(stateCols[0]);
-          ReadableColumn countCol = tb.asColumn(stateCols[1]);
-          double[] ret = new double[sumCol.size()];
-          for (int i = 0; i < sumCol.size(); ++i) {
-            ret[i] = ((double) sumCol.getLong(i)) / countCol.getLong(i);
-          }
-          return ImmutableList.of(new DoubleColumn(ret, ret.length));
-        }
-      case STDEV:
-        {
-          ReadableColumn sumCol = tb.asColumn(stateCols[0]);
-          ReadableColumn sumSquaredCol = tb.asColumn(stateCols[1]);
-          ReadableColumn countCol = tb.asColumn(stateCols[2]);
-          double[] ret = new double[sumCol.size()];
-          for (int i = 0; i < sumCol.size(); ++i) {
-            double first = ((double) sumSquaredCol.getLong(i)) / countCol.getLong(i);
-            double second = ((double) sumCol.getLong(i)) / countCol.getLong(i);
-            ret[i] = Math.sqrt(first - second * second);
-          }
-          return ImmutableList.of(new DoubleColumn(ret, ret.length));
-        }
-      default:
-        throw new IllegalArgumentException(aggOp + " is invalid");
-    }
-  }
-
-  @Override
   protected boolean isSupported(final AggregationOp aggOp) {
```

Comment:
Here the "supported" means the exposed aggregate such as AVG but not internal aggregates from 'getInternalOps()'. `addRow()` is only called with an `aggOp` from 'getInternalOps()' . Do you think we should remove the check? It's more like a check for RACO. I feel that it doesn't hurt to double-check here.